Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift

```diff
@@ -673,15 +698,32 @@ extension BrowserViewController: WKNavigationDelegate {
         guard let tab = tabManager[webView] else { return }
         searchTelemetry?.trackTabAndTopSiteSAP(tab, webView: webView)
         tab.url = webView.url
-        // When tab url changes after web content starts loading on the page
-        // We notify the contect blocker change so that content blocker status can be correctly shown on beside the URL bar
+
+        // Only update search term data with valid search term data
         let searchTerm = tab.tabGroupData.tabAssociatedSearchTerm
         let searchUrl = tab.tabGroupData.tabAssociatedSearchUrl
         let tabNextUrl = tab.tabGroupData.tabAssociatedNextUrl
         if !searchTerm.isEmpty, !searchUrl.isEmpty, let nextUrl = webView.url?.absoluteString, !nextUrl.isEmpty, nextUrl != searchUrl, nextUrl != tabNextUrl {
+            
+            if tab.adsTelemetryRedirectUrlList.count > 0,
+               !tab.adsProviderName.isEmpty,
+                tab.adsTelemetryUrlList.count > 0,
+               !tab.adsProviderName.isEmpty,
+                let startingRedirectHost = tab.startingSearchUrlWithAds?.host,
+                let lastRedirectHost = tab.adsTelemetryRedirectUrlList.last?.host,
+                lastRedirectHost != startingRedirectHost {
+                
+                AdsTelemetryHelper.trackAdsClickedOnPage(providerName: tab.adsProviderName)
+                tab.adsTelemetryUrlList.removeAll()
+                tab.adsTelemetryRedirectUrlList.removeAll()
+                tab.adsProviderName = ""
+            }
+
             tab.updateTimerAndObserving(state: .tabNavigatedToDifferentUrl, searchTerm: searchTerm, searchProviderUrl: searchUrl, nextUrl: nextUrl)
         }
 
+        // When tab url changes after web content starts loading on the page
+        // We notify the contect blocker change so that content blocker status can be correctly shown on beside the URL bar
```

Comment:
`contect` 
```suggestion
        // We notify the content blocker change so that content blocker status can be correctly shown on beside the URL bar
```